genie-web/src/main/java/com/netflix/genie/web/events/JobFinishedSNSPublisher.java

```diff
@@ -0,0 +1,211 @@
+    private static final String JOB_AGENT_VERSION_KEY_NAME = "job-agent-version";
+    private static final String JOB_GROUPING_INSTANCE_KEY_NAME = "job-grouping-instance";
+    private static final String JOB_STATUS_MESSAGE_KEY_NAME = "job-status-message";
+    private static final String JOB_API_CLIENT_HOSTNAME_KEY_NAME = "job-api-client-hostname";
+    private static final String JOB_REQUESTED_MEMORY_KEY_NAME = "job-requested-memory";
+    private static final String JOB_AGENT_HOSTNAME_KEY_NAME = "job-agent-hostname";
+    private static final String JOB_API_CLIENT_USER_AGENT_KEY_NAME = "job-api-client-user-agent";
+    private static final String JOB_EXIT_CODE_KEY_NAME = "job-exit-code";
+    private static final String JOB_NUM_ATTACHMENTS_KEY_NAME = "job-num-attachments";
+    private static final String JOB_ARCHIVE_LOCATION_KEY_NAME = "job-archive-location";
+    private static final String JOB_USED_MEMORY_KEY_NAME = "job-used-memory";
+    private static final String COMMAND_ID_KEY_NAME = "command-id";
+    private static final String COMMAND_NAME_KEY_NAME = "command-name";
+    private static final String COMMAND_VERSION_KEY_NAME = "command-version";
+    private static final String COMMAND_DESCRIPTION_KEY_NAME = "command-description";
+    private static final String COMMAND_CREATED_KEY_NAME = "command-created";
+    private static final String COMMAND_UPDATED_KEY_NAME = "command-updated";
+    private static final String COMMAND_EXECUTABLE_KEY_NAME = "command-executable";
+    private static final String CLUSTER_ID_KEY_NAME = "cluster-id";
+    private static final String CLUSTER_NAME_KEY_NAME = "cluster-name";
+    private static final String CLUSTER_VERSION_KEY_NAME = "cluster-version";
+    private static final String CLUSTER_DESCRIPTION_KEY_NAME = "cluster-description";
+    private static final String CLUSTER_CREATED_KEY_NAME = "cluster-created";
+    private static final String CLUSTER_UPDATED_KEY_NAME = "cluster-updated";
+    private static final String APPLICATIONS_KEY_NAME = "applications";
+
+    private final JobPersistenceService jobPersistenceService;
+
+    /**
+     * Constructor.
+     *
+     * @param snsClient             Amazon SNS client
+     * @param properties            configuration properties
+     * @param jobPersistenceService job persistence service
+     * @param registry              metrics registry
+     * @param mapper                object mapper
+     */
+    public JobFinishedSNSPublisher(
+        final AmazonSNS snsClient,
+        final SNSNotificationsProperties properties,
+        final JobPersistenceService jobPersistenceService,
+        final MeterRegistry registry,
+        final ObjectMapper mapper
+    ) {
+        super(properties, registry, snsClient, mapper);
+        this.jobPersistenceService = jobPersistenceService;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void onApplicationEvent(final JobStateChangeEvent event) {
```

Comment:
Would it be more valuable for the system to send an internal `JobFinishedEvent` (I believe we have something like this in v3 already) so that you don't have to do additional filtering in here? Plus in that event you could include information already and maybe save db calls?

Reply:
The DB call is unavoidable. 
Publishing that different event on the bus does indeed simplify this publisher but puts the same complexity upstream so overall i kinda like the current approach of listening to job status changes and doing whatever necessary